pkg/resource/deploy/step_generator.go

```diff
@@ -769,6 +769,31 @@ func (sg *stepGenerator) GenerateDeletes(targetsOpt map[resource.URN]bool) ([]St
 	return dels, nil
 }
 
+func (sg *stepGenerator) getChildrenOfTarget(target resource.URN) map[resource.URN]bool {
+	allTargets := make(map[resource.URN]bool)
+	allTargets[target] = true
+
+	var findChildren func(urn resource.URN)
+	findChildren = func(urn resource.URN) {
+		for _, res := range sg.deployment.prev.Resources {
+			if res.Parent != "" {
+				if _, has := allTargets[res.URN]; has {
+					// already included
+					continue
+				}
+
+				if _, has := allTargets[res.Parent]; has {
+					allTargets[res.URN] = true
+					findChildren(res.URN)
+				}
+			}
+		}
+	}
+	findChildren(target)
```

Comment:
I don't think that we need to do this recursively.

The list of resources is a topological sort of the reverse-dependency graph, so any resource R will appear in the list before its dependents and its children. This implies that given R, we can find all of its direct dependents and children via a linear scan of the list starting from R. By induction, we can find all of the transitive dependents and children of R via the same linear scan. So this can just be the following:

```suggestion
	for _, res := range sg.deployment.prev.Resources {
		if _, has := allTargets[res.Parent]; has {
			allTargets[res.URN] = true
		}
	}
```